### PR TITLE
Adding password recovery portal changes

### PR DESCRIPTION
**Pull request by lmanohara**
- Password recovery UI changes.
- Email mgt feature installation.
- Event handler notification feature installation.

### CODE REVIEW

File: portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js [hunk 1]
```diff
@@ -264,4 +264,27 @@ var recoveryManager = {};
         return getRecoveryConfigs();
     };
 
+    recoveryManager.sendRecoveryNotification = function (uniqueUserId) {
+        if (uniqueUserId) {
+            try {
+                var recoveryMgtService = callOSGiService("org.wso2.is.portal.user.client.api.RecoveryMgtService", "setPasswordRecoveryNotification", [uniqueUserId]);
+            } catch (e) {
+                //todo need show error message in UI
+                Log.error(e + " ");
```
Comment (ayshsandu) [inline portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js]: Log meaningful error message, if osgi service does error logging, then Log.error(e.message);

File: portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js [hunk 2]
```diff
@@ -264,4 +264,27 @@ var recoveryManager = {};
...
+        }
+    }
+
+    recoveryManager.updatePassword = function (code, password) {
+        try {
+            var passwordChar = Java.to(password.split(''), 'char[]');
+            var recoveryMgtService = callOSGiService("org.wso2.is.portal.user.client.api.RecoveryMgtService", "updatePassword", [code, passwordChar]);
+        } catch (e) {
+            //todo need show error message in UI
+            Log.error(e + " ");
```
Comment (ayshsandu) [inline portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js]: Log meaningful error message, if osgi service does error logging, then Log.error(e.message);

File: portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js (hunk 1 above)
Comment (lmanohara) [inline portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js]: fixed https://github.com/wso2/product-is/pull/1127/commits/519f3eaee63647cf17a269037d8f7abaf0050c80

File: portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js (hunk 2 above)
Comment (lmanohara) [inline portal/apps/org.wso2.carbon.iam.user-portal.app/src/main/modules/recovery-manager.js]: fixed https://github.com/wso2/product-is/pull/1127/commits/519f3eaee63647cf17a269037d8f7abaf0050c80

**Pull request merged**